R/lang.R

```diff
@@ -1,104 +1,144 @@
-#' Is an object a language object?
+#' Is an object an expression?
 #'
 #' These helpers are consistent wrappers around their base R
-#' equivalents. A language object is either an atomic vector
-#' (typically a scalar), a name (aka a symbol), a call, or a pairlist
-#' (used for function arguments).
-#'
-#' \code{is_literal()} is a predicate that returns \code{TRUE} for the
-#' subset of literals that are created by R when parsing text (see
-#' \code{\link{parse_expr}()}): numbers, strings and \code{NULL}.
-#' Along with symbols, these literals are the terminating nodes in a
-#' parse tree. Note that in the most general sense, a literal is any R
-#' object that evaluates to itself and that can be evaluated in the
-#' empty environment. For instance, \code{quote(c(1, 2))} is not a
-#' literal, but the result of evaluating it in
-#' \code{\link{base_env}()} is (in this case an atomic vector).
-#' Technically, this sort of literal objects can be inlined in
-#' language expressions. If your function accepts arbitrary
+#' equivalents. \code{is_expr()} tests for expressions, the set of
+#' objects that can be obtained from parsing R code. An expression
+#' can be one of two things: either a symbolic object (for which
+#' \code{is_symbolic()} returns \code{TRUE}), or a parsable literal
+#' (testable with \code{is_parsable_literal()}).
+#'
+#' Literals are the set of R objects that evaluate to themselves when
```

Comment:
I think you need to emphasise:

* that any R object can be placed in a call
* this doesn't happen normally (when parsing text)
* symbols and calls have special behaviour